ingest/bin/apply-geolocation-rules

```diff
@@ -113,24 +113,33 @@ def get_annotated_geolocation(geolocation_rules, raw_geolocation, rule_traversal
     if current_rules is None:
         # If we've used all general rules and we still haven't found a match,
         # then there are no applicable rules for this geolocation
-        if all(value == '*' for value in rule_traversal):
-          return None
-
-        # Find the index of the first of the consecutive '*' from the
-        # end of the rule_traversal
-        # [A, *, B, *] => first_consecutive_general_rule_index = 3
-        # [A, *, *, *] => first_consecutive_general_rule_index = 1
-        first_consecutive_general_rule_index = 0
-        for index, field_value in reversed(list(enumerate(rule_traversal))):
-            if field_value == '*':
-                first_consecutive_general_rule_index = index
-            else:
-                break
-
-        # Edit the raw value of the field directly before the first consecutive '*'
-        # in the rule traversal in hopes that by moving to a general rule on
-        # a higher level, we can find a matching rule.
-        rule_traversal[first_consecutive_general_rule_index - 1] = '*'
+        if (len(rule_traversal) == len(raw_geolocation) and
+            all(value == '*' for value in rule_traversal)):
+            return None
+
+        # If we failed to find matching rule with a general rule as the last
+        # traversal target, then delete all trailing '*'s to reset rule_traversal
+        # to end with the last index that is currently NOT a '*'
+        # [A, * , B, *] => [A, *, B]
+        # [A, B,  *, *] => [A, B]
+        # [A, *,  *, *] => [A]
+        if rule_traversal[-1] == '*':
+            # Find the index of the first of the consecutive '*' from the
+            # end of the rule_traversal
+            # [A, *, B, *] => first_consecutive_general_rule_index = 3
+            # [A, B, *, *] => first_consecutive_general_rule_index = 2
+            # [A, *, *, *] => first_consecutive_general_rule_index = 1
+            for index, field_value in reversed(list(enumerate(rule_traversal))):
+              if field_value == '*':
+                  first_consecutive_general_rule_index = index
+              else:
+                  break
+
+            rule_traversal = rule_traversal[:first_consecutive_general_rule_index]
+
+        # Set the final value to '*' in hopes that by moving to a general rule,
+        # we can find a matching rule.
+        rule_traversal[-1] = '*'
```

Comment:
This fails a test I drafted in #54:

https://github.com/nextstrain/monkeypox/blob/a2d8dcb876a9ad92fe900e4996785152671994f3/ingest/bin/tests/test_apply_geolocation_rules.py#L77-L83

```
>               rule_traversal[-1] = '*'
E               IndexError: list assignment index out of range
```

Reply:
Ah, this is why I need to move towards TDD...fixed up in latest push.

Reply:
Oh sure, go for it!

Reply:
(I reworded the commit message a little to try to clarify my reasoning for the change)